internal/github/github_test.go

```diff
@@ -297,201 +304,307 @@ func TestIsPRMergedOrClosed_Error(t *testing.T) {
-			checks: []*github.CheckRun{},
-			checks2: []*github.CheckRun{
-				{
-					Status:     github.String("completed"),
-					Conclusion: github.String("success"),
-				},
-			},
-			status: &github.CombinedStatus{
-				State:      github.String("pending"),
-				TotalCount: github.Int(0),
-			},
-			status2: &github.CombinedStatus{
-				State:      github.String("success"),
-				TotalCount: github.Int(1),
-			},
-			// For status which change from no result to anything else, we
-			// return unknown to get called again on a subsequent iteration.
-			expectedStatus: CIStatusUnknown,
+			name: "pending with checks and statuses",
+			mockGraphQL: `
+            {
+				"data": {
+					"repository": {
+						"object": {
+							"statusCheckRollup": {
+								"state": "PENDING",
+								"contexts": {
+									"nodes": [
+										{
+											"__typename": "CheckRun",
+											"name": "build",
+											"conclusion": "PENDING"
+										},
+										{
+											"__typename": "StatusContext",
+											"context": "deployment",
+											"state": "PENDING"
+										}
+									],
+									"pageInfo": {
+										"hasNextPage": false,
+										"endCursor": null
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+            `,
+			expectedStatus: CIStatusPending,
 		},
 		{
-			name: "failure",
-			status: &github.CombinedStatus{
-				State:      github.String("failure"),
-				TotalCount: github.Int(1),
-			},
-			expectedStatus: CIStatusFailed,
+			name: "no data",
+			mockGraphQL: `
+            {
+				"data": {
+					"repository": {
+						"object": null
+					}
+				}
+			}
+            `,
+			expectedStatus: CIStatusPassed,
 		},
 		{
-			name: "unknown",
-			status: &github.CombinedStatus{
-				State:      github.String("unknown"),
-				TotalCount: github.Int(1),
-			},
+			name: "unknown status",
+			mockGraphQL: `
+            {
+				"data": {
+					"repository": {
+						"object": {
+							"statusCheckRollup": {
+								"state": "INVALID_STATE",
+								"contexts": {
+									"nodes": [
+										{
+											"__typename": "CheckRun",
+											"name": "test-check",
+											"conclusion": "invalid"
+										},
+										{
+											"__typename": "StatusContext",
+											"context": "test-status",
+											"state": "unknown"
+										}
+									],
+									"pageInfo": { "hasNextPage": false, "endCursor": null }
+								}
+							}
+						}
+					}
+				}
+			}
+            `,
 			expectedStatus: CIStatusUnknown,
 		},
 	}
 
 	for _, tt := range tests {
-		tt := tt
-
 		t.Run(tt.name, func(t *testing.T) {
 			t.Parallel()
 
-			ctx := context.Background()
+			mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if r.Method == "POST" && r.URL.Path == "/graphql" {
+					w.Header().Set("Content-Type", "application/json")
+					_, _ = w.Write([]byte(tt.mockGraphQL))
+				} else {
+					http.NotFound(w, r)
+				}
+			}))
```

Comment:
this is a nit/styleish point. In my experience a lot of Go developers write code so that the "normal" path is outside of any `if` blocks. A good rule of thumb is to try to keep it as left aligned as possible, and it also usually means you don't need an `else` part to your condition. Here it would be like this (sorry for any typos, I just did this in the editor here rather than trying it):

```suggestion
				if r.Method != "POST" || r.URL.Path != "/graphql" {
					http.NotFound(w, r)
					return
				}

				w.Header().Set("Content-Type", "application/json")
				_, _ = w.Write([]byte(tt.mockGraphQL))
			}))
```

See what I mean? The unusual/error/less common path is indented, and the normal one is aligned to the left. It means that people reading your code get a nice visual cue as to the flow just by looking at the shape of the text. 🙂

Reply:
Fixed. Much easier to read, thanks.